sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -183,43 +187,68 @@ Table 2. Terminology.
-operation on the device.
-
 | Edge
 | Dependency between commands as a happens-before relationship.
 
-When recording a queue to construct a graph, an edge in the SYCL graph represents
-a data dependency between two nodes. These dependencies are expressed by the user
-code through buffer accessors. There is also the partial ability to track USM
-data dependencies provided the pointers used in the graph nodes are the same.
-With the limitation that a node taking an offset USM pointer input will not be
-identified as having an edge to another node taking a pointer input to the base
-address of the same USM allocation.
+|===
+
+==== Explicit Graph Building API
+
+When using the explicit graph building API to construct a graph, nodes and
+edges are captured as follows.
+
+Table 3. Explicit Graph Definition.
+[%header,cols="1,3"]
+|===
+| Concept | Description
+
+| Node
+| In the explicit graph building API nodes are created by the user invoking
+methods on a modifiable graph. Each node represent either a command-group
+function, empty operation, or device memory allocation/free.
+
+| Edge
+| In the explicit graph building API edges are defined by the user. This is
+either through buffer accessors, the `make_edge()` free function, or by passing
+dependent nodes on creation of a new node.
+|===
+
+==== Queue Recording API
+
+When using the record & replay API to construct a graph by recording a queue,
+nodes and edges are captured as follows.
 
-In the explicit graph building API, `make_edge()` is used to define the dependency
-rather than inferring them from data dependencies.
+Table 4. Recorded Graph Definition.
+[%header,cols="1,3"]
+|===
+| Concept | Description
+
+| Node
+| Nodes in a queue recorded graph represent each of the command group
+submissions of the program. Each submission encompasses either one or both of
+a.) some data movement, b.) a single asynchronous kernel launch. Nodes cannot
+define forward edges, only backwards (i.e. kernels can only create dependencies
+on things that have already happened). This means that transparently a node can
+depend on a previously recorded graph (sub-graph), which works by creating edges
+to the individual nodes in the old graph. Explicit memory operations without
+kernels, such as a memory copy, are still classed as nodes under this
+definition, as the {explicit-memory-ops}[SYCL 2020 specification states] that
+these can be seen as specialized kernels executing on the device.
+
+| Edge
+| An edge in a queue recorded graph represents a data dependency between two
+nodes. These dependencies are expressed by the user code through buffer
+accessors. There is also the partial ability to track USM data dependencies
+provided the pointers used in the graph nodes are the same. With the limitation
+that a node taking an offset USM pointer input will not be identified as having
+an edge to another node taking a pointer input to the base address of the same
+USM allocation.
```

Comment:
I wonder about including this wording because in some ways it seems like a implementation detail or limitation. In theory I think a runtime should be able to track usm allocations, though I'm not sure if DPC++ does that currently so it may be a general limitation in practice.

Reply:
That's a good point, this wording is based on a limitation in the ComputeCpp prototype which I don't think is a general limitation. I think if we devoted engineering resources to looking into this, we'd be able to track the USM data dependencies properly in an implementation. I'll change this wording.